src/main/java/org/jitsi/impl/reservation/rest/RESTReservations.java

```diff
@@ -217,7 +216,7 @@ else if (result.statusCode == 409
     /**
      * Deletes conference for given <tt>mucRoomName</tt> through the API.
      */
-    Result deleteConference(String mucRoomName)
+    private synchronized Result deleteConference(String mucRoomName)
```

Comment:
My thinking is: The method requires the lock, so it should be annotated. The fact that it is currently only called from places which already hold the lock is incidental. Specifically, I have two problems with the missing lock:
1. It is harder to verify that the code is safe (I had to follow the usage). 
2. It is a lot easier to screw it up (in a subtle and hard to find manner) when adding new code.